src/Executor/Run.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Diagnostics;
+
+namespace Microsoft.DotNet.Execute
+{
+    public class Run
+    {
+        private System.Diagnostics.Process _process;
+        
+        public int ExecuteProcess(string filename, string args = null)
+        {
+            try
+            {
+                var psi = new ProcessStartInfo
+                {
+                    FileName = filename,
+                    Arguments = args,
+                    RedirectStandardOutput = true,
+                    UseShellExecute = false
+                };
+
+                _process = new System.Diagnostics.Process();
+                _process.StartInfo = psi;
+
+                // Set our event handler to asynchronously read the output.
+                _process.OutputDataReceived += new DataReceivedEventHandler(ReadOutputHandler);
+
+                _process.Start();
+                _process.BeginOutputReadLine();
+
+                _process.WaitForExit();
```

Comment:
> I feel like I've seen some race conditions with output and the exit event. Sometimes the exit event will happen while there is still pending output. If this process exits at that point, the "end" of the output will get lost.

I'll take a look and as we talked look at Async functions

> This also doesn't redirect the Error stream, (because it's a totally different, needing RedirectStandardError and ErrorDataRecieved).

Msbuild redirects everything to the input one so I didn't use the Error stream, but as you pointed out when we talked we need the Error stream too for other processes different than msbuild.I'll change it.

>Also, what about input?

I consider that this scenario doesn't apply for the executor.